### PR TITLE
WIP: add warning for importing JSONio lib for matlab/octave

### PR DESCRIPTION
**Pull request by Remi-Gau**
Regarding the issue #119
This a PR equivalent to the closed PR 46 but without the line ending problems

I am tempted to keep the original call to the function jsonwrite and simply add a warning if it fails with a link to the library.

Other possibility: call the bids.util.jsonencode function from the matlab-bids library which would also make sense because this is a bids related issue.

Any thought? @gllmflndn

Once we have decided I will change all the example scripts accordingly.

### CODE REVIEW

File: matlabCode/createBIDS_anat_Full_T1w_json.m [hunk 1]
```diff
@@ -1,6 +1,8 @@
 %% Template Matlab script to create an BIDS compatible sub-01_ses-01_acq-FullExample_run-01_T1w.json file
 % This example lists all required  RECOMMENDED and optional fields.
 % When adding additional metadata please use CamelCase 
+%
+% Writing json files relies
```
Comment (KirstieJane) [inline matlabCode/createBIDS_anat_Full_T1w_json.m]: Is this sentence incomplete?

File: matlabCode/createBIDS_anat_Full_T1w_json.m [hunk 2]
```diff
@@ -241,11 +243,23 @@
...
+% this makes the json look pretier when opened in a txt editor
+json_options.indent = '    '; 
 
 jsonSaveDir = fileparts(anat_json_name);
 if ~isdir(jsonSaveDir)
     fprintf('Warning: directory to save json file does not exist, create: %s \n',jsonSaveDir)
 end
 
-jsonwrite(anat_json_name,anat_json,json_options)
+try
```
Comment (KirstieJane) [inline matlabCode/createBIDS_anat_Full_T1w_json.m]: I can't properly review the code (I don't know matlab) but it looks good to me!

File: matlabCode/createBIDS_anat_Full_T1w_json.m [hunk 3]
```diff
@@ -241,11 +243,23 @@
...
+    % bids.util.jsonencode(anat_json_name, anat_json)
+    jsonwrite(anat_json_name,anat_json,json_options)
+catch
+    str = sprintf('%s\n%s\n%s\n%s',...
+        'Writing the JSON file seems to have failed.', ...
+        'Make sure that the following libraries are in the matlab/octave path:', ...
```
Comment (dorahermes) [inline matlabCode/createBIDS_anat_Full_T1w_json.m]: It reads now as if you need both libraries, do you only need either?

**Comment by dorahermes** [comment]
Thanks!!! 
Are the jsonwrite/jsonread the same for the two repositories that you suggest?

File: matlabCode/createBIDS_anat_Full_T1w_json.m (hunk 1 above)
Comment (Remi-Gau) [reply matlabCode/createBIDS_anat_Full_T1w_json.m]: Yup... My train of thought derailed. Was about to start putting some comments on which library need to be used for the script to work but I figured that we should first agree on that before putting the comments.

**Comment by Remi-Gau** [comment]
Well JSONio should be enough and should cover matlab and octave and does use a function called `jsonwrite`

BIDS-matlab checks which library is available (spm, JSONio, built-in: in that order) and uses the relevant function with the appropriate syntax. So in practice having BIDS-matlab only as a library would only cover matlab users with a version > 2016 if I am correct. 

So in that respect option JSONio only would be simpler but less "BIDSy", though I suspect we don't want users to HAVE to use more tools than they need.

**Comment by Remi-Gau** [comment]
I did some auto indent on matlab (ctrl + I) and it removed a bunch of trailing spaces: sorry for the confusion.

Also added some `;` to suppres output to the console.

Just realized that there are some weird character encoding happening in the comments in the eeg script. Should I modify them?

File: matlabCode/createBIDS_anat_Full_T1w_json.m [hunk 4]
```diff
@@ -241,11 +245,20 @@
 
 
 %% Write JSON
-json_options.indent = '    '; % this makes the json look pretier when opened in a txt editor
+% this makes the json look pretier when opened in a txt editor
```
Comment (gllmflndn) [inline matlabCode/createBIDS_anat_Full_T1w_json.m]: pret**t**ier.
(this option is specific to JSONio and will not work with jsonencode).

Comment (Remi-Gau) [reply matlabCode/createBIDS_anat_Full_T1w_json.m]: I think that for now we will stick with JSONio so this should be fine, unless we have some very good reason not to.

**Comment by dorahermes** [comment]
This is great, thanks! @Remi-Gau is it ready to merge?

**Comment by Remi-Gau** [comment]
Now it is! :-)

**Comment by Remi-Gau** [comment]
While I am at it do you want me to take care of issue #112 ?

**Comment by robertoostenveld** [comment]
The last comment is 2 months old. Could this please be merged into the master?

**Comment by Remi-Gau** [comment]
@all-contributors please add @gllmflndn  for review, ideas

**Pull request merged**